### PR TITLE
Update git notes re: rebasing and merging

### PR DESCRIPTION
**Pull request by mattdoller**

### CODE REVIEW

File: git.md [hunk 1]
```diff
@@ -26,6 +30,82 @@
...
+|master ✓|$ git checkout feature
+|feature ✓|$ git rebase master
+
+A--F--G--H--I                 (master)
+             \
+              \
+               B'--C'--D'--E' (feature)
+```
+
+Note that the commits B, C, D, and E have been replace by B', C', D', E'.  The
```
Comment (schmidtg) [inline git.md]: 'replaced'

File: git.md [hunk 2]
```diff
@@ -26,6 +30,82 @@
...
+commit.  You will need to issue a `git commit`, and update the commit message.
+Git will include all of the intermediate commit messages that you have made
+for your feature branch, but you should cut this down to a single message
+using the standard commit message format (50-character first line, blank
+line, then more descriptive paragraphs limited to 72-character lines).
+
+Now commit J contains all of the functionality of the feature branch.
+Rebasing branches and squash merges help keep the master history clean.
+
+Note that when you rebase your branch, you are effectively creating replacing
```
Comment (schmidtg) [inline git.md]: Can remove 'creating' here

**Comment by schmidtg** [comment]
:shipit: 

If you want to get fancy, it'd be nice to have a visual example of a squashed merge commit message (with a headline *BAD*), and how we want to avoid those (with a headline *GOOD*).

**Comment by mattdoller** [comment]
closing this - i had it done against a personal fork of the sermo guides repo.  going to re-post as a branch on the sermo repo

**Pull request closed without merging**